pattern_import_export_xlsx/tests/test_pattern_export.py

```diff
@@ -0,0 +1,230 @@
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        values = {"export_format": "xlsx"}
+        cls.ir_exports.write(values)
+        cls.ir_exports_m2m.write(values)
+
+    def test_generate_pattern_with_basic_fields(self):
+        self.ir_exports.pattern_last_generation_date = False
+        self.ir_exports.pattern_file = False
+        self.ir_exports.export_fields[0].unlink()
+        self.ir_exports.export_fields[2:].unlink()
+        res = self.ir_exports.generate_pattern()
+        self.assertEqual(res, True)
+        self.assertNotEqual(self.ir_exports.pattern_file, False)
+        self.assertNotEqual(self.ir_exports.pattern_last_generation_date, False)
+        decoded_data = base64.b64decode(self.ir_exports.pattern_file)
+        wb = open_workbook(file_contents=decoded_data)
+        sheet1 = wb.sheet_by_index(0)
+        self.assertEqual(sheet1.name, "Partner list")
+        self.assertEqual(sheet1.cell_value(0, 0), "name")
+        self.assertEqual(sheet1.cell_value(0, 1), "street")
+
+    def test_generate_pattern_with_many2one_fields(self):
+        self.ir_exports.export_fields[0:3].unlink()
+        self.ir_exports.export_fields[1].unlink()
+        self.ir_exports.generate_pattern()
+        decoded_data = base64.b64decode(self.ir_exports.pattern_file)
+        wb = open_workbook(file_contents=decoded_data)
+        self.assertEqual(len(wb.sheets()), 2)
+        sheet1 = wb.sheet_by_index(0)
+        self.assertEqual(sheet1.cell_value(0, 0), "country_id")
+        sheet2 = wb.sheet_by_index(1)
+        self.assertEqual(sheet2.name, "Country list (code)")
+        self.assertEqual(sheet2.cell_value(1, 0), "BE")
+        self.assertEqual(sheet2.cell_value(2, 0), "FR")
+        self.assertEqual(sheet2.cell_value(3, 0), "US")
+
+    def _check_is_many2many(self, export_lines):
+        """
+        Ensure the is_many2many is correct for given export lines.
+        """
+        for export_line in export_lines:
+            expected_result = False
+            if (
+                export_line.is_many2x
+                and export_line.export_id.resource
+                and export_line.name
+            ):
+                field, model = export_line._get_last_field(
+                    export_line.export_id.resource, export_line.name
+                )
+                if self.env[model]._fields[field].type == "many2many":
+                    expected_result = True
+            if expected_result:
+                self.assertTrue(export_line.is_many2many)
+            else:
+                self.assertFalse(export_line.is_many2many)
+
+    def test_generate_pattern_is_many2many(self):
+        """
+        Ensure the is_many2many boolean field on the ir.exports.line
+        is correctly set.
+        """
+        self._check_is_many2many(self.ir_exports.export_fields)
+        self._check_is_many2many(self.ir_exports_m2m.export_fields)
+
+    def test_generate_pattern_with_many2many_fields1(self):
```

Comment:
Just a point of view.
I try to do smaller test (20 lines is enough for my brain ) and I try to avoid testing twice the same feature.

I would prefer to have 3 test instead of 2 

```
def test_m2m_select_tab(self):
    # test that the content of the second tab is correct, we have the list of company

def test_m2m_one_occurence(self):
    # check that we have 1 col (with |)

def test_m2m_one_occurence(self):
    # check that we have 2 col (with |)
```

What do you think ?